### PR TITLE
types: add DateString type

### PR DESCRIPTION
**Pull request by kjdelisle**
connected to https://github.com/strongloop/loopback-connector-mysql/issues/149

### CODE REVIEW

**Comment by kjdelisle** [comment]
The definition this documentation currently refers to hasn't been written yet, please hold off on merging this until I post an update.

**Comment by kjdelisle** [comment]
@crandmck Actually, Rand, could I get you to review the JSDoc portion of this PR?
https://github.com/strongloop/loopback-datasource-juggler/pull/1356
I want to make sure that the JSDoc will parse nicely. :)

File: pages/en/lb3/LoopBack-types.md
Comment (dhmlau) [inline pages/en/lb3/LoopBack-types.md:70]: will we have an anchor for #datestring in the apidoc?

Comment (jannyHou) [inline pages/en/lb3/LoopBack-types.md:70]: There is no tag with id `datestring` on that page. Other than that LGTM

**Review by jannyHou — APPROVED** [review]

File: pages/en/lb3/LoopBack-types.md
Comment (b-admike) [inline pages/en/lb3/LoopBack-types.md:70]: Right now, this link doesn't exist. I presume there is a PR somewhere for it?

**Review by ssh24 — APPROVED** [review]

**Review by loay — APPROVED** [review]
LGTM

**Review by b-admike — APPROVED** [review]

File: pages/en/lb3/LoopBack-types.md
Comment (crandmck) [reply pages/en/lb3/LoopBack-types.md:70]: I assume that's what @kjdelisle is talking about re https://github.com/strongloop/loopback-datasource-juggler/pull/1356.  Once that lands (with some modifications) _and_ juggler is published to npm, then the link should work.

**Comment by crandmck** [comment]
Looks fine, but as noted above, we should not land this until the change to juggler is published to npm, since that's when the feature will become broadly available and the links to the API docs will work as well.

**Pull request merged**
